### ISSUE TITLE
[Feature Request] Add support to volumes

### ISSUE THREAD

**Issue by rene** [linuxkit/linuxkit#4053]
The idea of this feature request (discussed with @deitch) is to propose the volume support like implemented by docker. Supporting volumes would allow a smart file sharing across different containers, for instance, common files needed by two different containers at _onboot_ and _services_ section could be provided to both (without any needs of bind mounts, copy files, rootfs shared permissions, etc) defining something like the following spec:

```yaml
volumes:
- image: source/image-with-data
  name: volA
onboot:
- image: some/containerA
  volume: volA:/opt/pathA
services:
- image: some/containerB
  volume: volA:/opt/pathB
```

The image with common data (specified in the volume volA) would then be mounted on the specified paths for containerA and containerB...

**Comment by deitch** [comment]
So the `volumes` section is just defining an arbitrary (but unique) name. Linuxkit determines where to put that in the filesystem, but it should be read-write, so that it can be read and written from other containers, whether onboot, services or onshutdown.

* Should its read-writeabiltiy be determined by some flag? e.g.
```yaml
volumes:
- image: source/image-with-data
  name: volA
  write: true
```
otherwise it is read-only?
* based on your description, there is an assumption that it is filled by an OCI image with whatever data is in it. In Docker, volumes normally are empty, i.e. it is just a shared space. What would be the syntax for both? Is the assumption that, if there is no image, it is just a blank directory?
* names must be unique, right? Otherwise we have conflicts, is that correct?

**Comment by rene** [comment]
> So the `volumes` section is just defining an arbitrary (but unique) name. Linuxkit determines where to put that in the filesystem, but it should be read-write, so that it can be read and written from other containers, whether onboot, services or onshutdown.
> 
>  Should its read-writeabiltiy be determined by some flag? e.g.
> 
> 
> ```yaml
> volumes:
> - image: source/image-with-data
>   name: volA
>   write: true
> ```
> 
> otherwise it is read-only?
> 

Yes, or maybe the other way around, have a flag for read-only, like `readonly: false`

>  based on your description, there is an assumption that it is filled by an OCI image with whatever data is in it. In Docker, volumes normally are empty, i.e. it is just a shared space. What would be the syntax for both? Is the assumption that, if there is no image, it is just a blank directory?

I was more interested in share image contents, so TBH I didn't think about blank directories, but it makes sense for me, but maybe we could enforce a name, perhaps defining a `type: blank` and use another field (instead of image) for the name... otherwise how it's gonna work? Create an arbitrary name and mount on all specified containers?

>    names must be unique, right? Otherwise we have conflicts, is that correct?

Yes

**Comment by deitch** [comment]
Fixed with #4065

**Comment by rene** [comment]
Thanks a lot @deitch , closing this issue since the feature was added to linuxkit.
